src/fn/edit-contacts.js

```diff
@@ -0,0 +1,188 @@
+const pouch = require('../lib/db');
+const toDocs = require('./csv-to-docs');
+const EDIT_RESERVED_COL_NAMES = [ 'parent', '_id', 'name', 'reported_date' ];
+const DOC_TYPES = ['district_hospital', 'health_center', 'clinic', 'person', 'user', 'user-settings', 'contact'];
+const DOCUMENT_ID =  'documentID';
+
+const execute = () => {
+  const args = parseExtraArgs(environment.pathToProject, environment.extraArgs);
+  const db = pouch();
+  const docDirectoryPath = args.docDirectoryPath;
+  fs.mkdir(docDirectoryPath);
+  const saveJsonDoc = doc => fs.write(`${docDirectoryPath}/${doc._id}.doc.json`, toDocs.toSafeJson(doc) + '\n');
+
+  const csvDir = `${environment.pathToProject}/csv`;
+  if(!fs.exists(csvDir)) {
+    warn(`No csv directory found at ${csvDir}.`);
+    return Promise.resolve();
+  }
+  const csvFiles = args.csvFiles;
+
+  return csvFiles.map(fileName => `${csvDir}/${fileName}`)
+    .filter(name => name.endsWith('.csv'))
+    .reduce((promiseChain, csv) =>
+      promiseChain
+      .then(() => {
+        info('Processing CSV file:', csv, '…');
+
+        const nameParts = fs.path.basename(csv).split('.');
+        const prefix = nameParts[0];
+        switch(prefix) {
+          case 'contact':  return processDocuments(csv, getIDs(csv, prefix), db, args);
+          case 'users': return processUsers(csv, getIDs(csv, prefix), db, args);
+          default: throw new Error(`Unrecognised CSV type ${prefix} for file ${csv}`);
+        }
+      })
+      .then(docs => addToModel(csv, docs)),
+      Promise.resolve())
+
+    .then(() => model.exclusions.forEach(toDocs.removeExcludedField))
+    .then(() => Promise.all(Object.values(model.docs).map(saveJsonDoc)));
+};
+
+const model = {
+  csvFiles: {},
+  docs: {},
+  exclusions: []
+};
+
+const addToModel = (csvFile, docs) => {
+  csvFile = csvFile.match(/^(?:.*[\/\\])?csv[\/\\](.*)\.csv$/)[1]; // eslint-disable-line no-useless-escape
+  model.csvFiles[csvFile] = docs;
+  docs.forEach(doc => {
+    model.docs[doc._id] = doc;
+  });
+};
+
+function getIDs(csv, docType) {
+  const { rows, cols } = fs.readCsv(csv);
+  const index = cols.indexOf(DOCUMENT_ID);
+  if (index === -1){
+   throw Error('missing "documentID" column.');
+  }
+
+  const idPrefix =  docType === 'contact' ? '' : 'org.couchdb.user:';
+  return rows.map((item) => idPrefix + item[index]);
+}
+
+function processDocs(docType, csv, documentDocs, args) {
+  const { rows, cols } = fs.readCsv(csv);
+  const uuidIndex = cols.indexOf(DOCUMENT_ID);
+  const colNames = args.colNames;
+  let toIncludeColumns, toIncludeIndex;
+  if (!colNames.length) {
+    warn(' No columns specified, the script will add all the columns in the CSV!');
+    toIncludeColumns = cols;
+    toIncludeIndex = [];
+
+  } else {
+    if (!columnsAreValid(cols,colNames)) {
+      throw Error('The column name(s) specified do not exist.');
+    }
+
+    toIncludeColumns = cols.filter(column => toIncludeColumns.includes(column.split(':')[0]));
```

Comment:
This line throws an error because you're calling `includes` for an `undefined` value.

Reply:
Nice catch, This should now be fixed.